noxfile.py

```diff
@@ -390,11 +391,23 @@ def release_build_bdist(session: nox.Session) -> None:
     # Pin a version to ensure reproducible builds.
     session.run("pip", "install", f"cibuildwheel=={cibuildwheel_version}")
 
+    flags = " ".join(
+        [
+            "-O2",  # O3 is unnecessary with the virtual-call-heavy code the GPI is and may bloat binary size
+            "-g1",  # Include only minimal debug information
```

Comment:
I have rarely seen `-g1` in use. I would just keep it at `-g` for simplicity and since we (as you also said) don't really care about file size here.

Reply:
```suggestion
            "-g",  # Include debug information to allow users to debug
```